sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.VisitDecl.cs

```diff
@@ -2748,204 +2748,282 @@ void VisitBitfieldDecl(FieldDecl fieldDecl, BitfieldDesc[] bitfieldDescs, Record
             _outputBuilder.BeginField(in desc);
             _outputBuilder.WriteRegularField(typeName, escapedName);
             _outputBuilder.BeginBody();
-            _outputBuilder.BeginGetter(_config.GenerateAggressiveInlining, isReadOnly: !Config.GenerateCompatibleCode);
-            var code = _outputBuilder.BeginCSharpCode();
-
-            code.WriteIndented("return ");
 
             var recordDeclName = GetCursorName(recordDecl);
 
-            var isSmallType = currentSize < 4;
-            var isRemappedToSelf = _config.RemappedNames.TryGetValue(typeName, out var remappedTypeName) && typeName.Equals(remappedTypeName, StringComparison.Ordinal);
-            var isTypeMismatch = type != builtinTypeBacking;
-            var isUnsignedToSigned = !isTypeBackingSigned && isTypeSigned;
+            // Small types become uint32/int32s after shifting
+            var isSmallType = fieldDecl.Type.Handle.SizeOf < 4;
+            var isSmallTypeBacking = currentSize < 4;
 
-            var needsCast = isSmallType || isRemappedToSelf || isTypeMismatch || isUnsignedToSigned;
-            var needsParenFirst = !isSmallType && isUnsignedToSigned;
-            var needsParenSecond = !needsParenFirst || isRemappedToSelf;
+            // Check if field/backing types match
+            var isTypeMismatch = type != typeBacking;
```

Comment:
Why the comparison against `typeBacking` rather than `buildinTypeBacking`? Won't that be meaningful for certain typedefs or enums?